pipelines/pro_image.go

```diff
@@ -0,0 +1,84 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"dagger.io/dagger"
+	"github.com/grafana/grafana-build/containers"
+)
+
+func ProImage(ctx context.Context, dc *dagger.Client, args PipelineArgs) error {
+	if len(args.PackageInputOpts.Packages) > 1 {
+		return fmt.Errorf("only one package is allowed: packages=%+v", args.PackageInputOpts.Packages)
+	}
+	packages, err := containers.GetPackages(ctx, dc, args.PackageInputOpts, args.GCPOpts)
+	if err != nil {
+		return fmt.Errorf("getting packages: packages=%+v %w", args.PackageInputOpts.Packages, err)
+	}
+
+	debianPackageFile := packages[0]
+
+	log.Printf("Cloning hosted Grafana...")
+	hostedGrafanaRepo, err := containers.CloneWithGitHubToken(dc, args.ProImageOpts.GitHubToken, "https://github.com/grafana/hosted-grafana.git", "main")
+	if err != nil {
+		return fmt.Errorf("cloning hosted-grafana repo: %w", err)
+	}
+
+	socketPath := "/var/run/docker.sock"
+	socket := dc.Host().UnixSocket(socketPath)
+
+	hostedGrafanaImage := fmt.Sprintf("%s/hosted-grafana-pro:%s", args.ProImageOpts.ContainerRegistry, args.ProImageOpts.ImageTag)
+
+	log.Printf("Building hosted Grafana image: %s", hostedGrafanaImage)
+	container := dc.Container().From("google/cloud-sdk:433.0.0-alpine").
+		WithExec([]string{
```

Comment:
Just one note (and I think it's a tradeoff for this PR): This will authenticate the host Docker Engine. You can create a DIND setup thanks to services in Dagger now that could get around this so that no credentials are leaked outside of this operation.

For this PR I think this is ok, though 🙂